pancancer_evaluation/data_models/tcga_data_model.py

```diff
@@ -232,6 +232,47 @@ def process_data_for_gene_id(self,
             self.y_test_df.status = np.random.permutation(
                 self.y_test_df.status.values)
 
+    def process_train_data_for_gene(self,
+                                    train_gene,
+                                    train_classification,
+                                    output_dir,
+                                    test_cancer_type=None,
+                                    shuffle_labels=False):
+        """
+        Prepare to train model on a given gene. Preparation of test data must
+        be done later (e.g. by calling another process_* function).
+
+        Arguments
+        ---------
+        train_identifier (str): gene combination to train on
+        train_classification (str): 'oncogene' or 'TSG' for the training gene
```

Comment:
I'm not sure what this training gene is? Is this a "negative control gene"?

Reply:
Ah, sorry, this was a copy/paste typo (should say `train_gene` rather than `train_identifier`).

This function separates the data into train/test datasets for the "pan-cross cancer" experiments. So the idea is to train the model on all cancer types but one, and test the model on the remaining cancer type. `train_gene` here just tells this data processing function which gene to get the labels for (and consequently which cancer types to use).

Reply:
Oh, wait - are you asking what TSG means here? It's an abbreviation for "tumor suppressor gene", not a particular gene. I'll write that out here since it definitely won't be clear to non-cancer people.

Reply:
you answered correctly, I thought "TSG" was an abbv for "training gene" thus the phrasing of my question.